### ISSUE TITLE
incorrect display of metrics for model.eval_metrics or when training with custom_metric with tweedie loss function

### ISSUE THREAD

**Issue by dubovikmaster** [catboost/catboost#2811]
The following simple example demonstrates that when using `Tweedie` as a loss function, metrics are not calculated correctly during model validation. The problem is apparently that metrics are calculated for `prediction_type='RawFormulaVal'` while by default in the predict method for the `Tweedie` loss function, `prediction_type='Exponent'` is used

I use catboost V1.2.7 with python 3.11

```python
from catboost import (
    CatBoostRegressor, 
    Pool
)
from sklearn.datasets import make_regression
from sklearn import metrics


X, y = make_regression(n_samples=100000, n_features=10, noise=1, random_state=42)
X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

train_pool = Pool(X_train, y_train)
eval_pool = Pool(X_test, y_test)

model = CatBoostRegressor(loss_function='Tweedie:variance_power=1.1', task_type='CPU', custom_metric=['R2', 'RMSE'], verbose=False)

model.fit(train_pool, plot=True)

model.eval_metrics(eval_pool, ['R2', 'RMSE'], ntree_start=999)

>>>  {'R2': [-0.036825647660097216], 'RMSE': [198.31499748613425]}

# this is incorrect
metrics.r2_score(y_test, model.predict(X_test, prediction_type='RawFormulaVal'))

>>>  -0.036825
# this is correct
metrics.r2_score(y_test, model.predict(X_test, prediction_type='Exponent'))
>>> -2.387022025135171e+121


```

Thus, what we see on the graph at the time of training and what the eval_metrics method produces is obtained with prediction_type='RawFormulaVal' instead of prediction_type='Exponent'
